src/test/java/com/puppycrawl/tools/checkstyle/internal/utils/ConfigurationUtil.java

```diff
@@ -44,6 +44,7 @@ public static Configuration loadConfiguration(String path) throws CheckstyleExce
         props.setProperty("checkstyle.regexp.header.file", "file");
         props.setProperty("checkstyle.importcontrol.file", "file");
         props.setProperty("checkstyle.importcontroltest.file", "file");
+        props.setProperty("checkstyle.java.version", "value");
```

Comment:
I feel you made this invalid by not using a number. If this doesn't fail when it loads the CS config in the test, then I would want to know why as that just questions why we then need to populate it.

Reply:
I am not sure but maybe when the parser encounters `${checkstyle.java.version}`, it looks for a corresponding entry in the properties list. If it doesn't find one, it fails with the error (`unable to parse configuration stream
 Property ${checkstyle.java.version} has not been set`) to enforce that all placeholders must be populated so that the configuration can be successfully loaded. if a placeholder like` ${checkstyle.java.version} `is found in the configuration file but isn't populated in the properties passed to` loadConfiguration()`, the parser can't resolve the placeholder, leading to the error.
 
However, even though the property setting is required to prevent the error, setting it to an invalid value like "value" won't cause an error because we are not actually loading the modules (checks) here. we are just parsing the config stream to test what tokens are missing from `checkstyle-checks.xml`

Reply:
Ok thanks, that explained it. It isn't until `rootModule.configure` for when we start creating checks, and assigning properties. That is where it should fail on an invalid property value.